src/ChartInternal/Axis/Axis.ts

```diff
@@ -737,29 +737,20 @@ class Axis {
 	 */
 	getXAxisPadding(tickCount) {
 		const $$ = this.owner;
-		let padding = $$.config.axis_x_padding;
-
-		if (isEmpty(padding)) {
-			padding = {left: 0, right: 0};
-		} else {
-			padding.left = padding.left || 0;
-			padding.right = padding.right || 0;
-		}
+		let {left = 0, right = 0} = $$.config.axis_x_padding;
```

Comment:
I don't think that works for a number. It will assign `0` to both values (as `number.left` is `undefined` instead of assigning the number

Reply:
The right fix is to use `isNumber($$.config.axis_x_padding)` like in `getPadding`

Reply:
yeah, you're right. It needs to handle number value also. 
I'll reopen the issue to fix that. Thanks for the comment.